azurerm/data_source_eventhub_namespace.go

```diff
@@ -0,0 +1,112 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceEventHubNamespace() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceEventHubNamespaceRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"location": locationForDataSourceSchema(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"capacity": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"auto_inflate_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"maximum_throughput_units": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"default_primary_connection_string": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"default_secondary_connection_string": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"default_primary_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"default_secondary_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceEventHubNamespaceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).eventHubNamespacesClient
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	resp, err := client.Get(resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on EventHub Namespace %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("sku", string(resp.Sku.Name))
+	d.Set("capacity", resp.Sku.Capacity)
+
+	keys, err := client.ListKeys(resourceGroup, name, eventHubNamespaceDefaultAuthorizationRule)
+	if err != nil {
+		log.Printf("[ERROR] Unable to List default keys for EventHub Namespace %q (Resource Group %q): %+v", name, resourceGroup, err)
```

Comment:
Is this worth error-ing out completely instead of just logging? The `ERROR` makes it seem like it's an issue instead of keys just not being there. Might be worth swapping to an `INFO` or `WARN` tag.

Reply:
good shout - I'll update this to a `WARN` 👍